File: MobileApp/AlexaDeviceFinder/AlexaDeviceFinder.Android/Utilities/PreferencesManager.cs
using Android.Content;
using DeviceFinder.Droid.Notifications;

namespace DeviceFinder.Droid.Utilities
{
    public class PreferencesManager
    {
        private string PREFERENCES_NAME = "com.Jamespfluger.Alexadevicefinder.SHARED_PREFERENCES";
        private string PREFERENCE_NAME_FIREBASE = "firebasetoken";
        private string PREFERENCE_NAME_AMAZON_USER_ID = "amazonuserid";
        private string PREFERENCE_NAME_ALEXA_USER_ID = "alexauserid";
        private string PREFERENCE_NAME_DEVICE_NAME = "devicename";
        private ISharedPreferences Preferences;
        private Context Context;

        public PreferencesManager(Context context)
        {
            this.Preferences = context.GetSharedPreferences(PREFERENCES_NAME, FileCreationMode.Private);
            this.Context = context;
        }

        public string GetDeviceId()
        {
            string firebaseToken = Preferences.GetString(PREFERENCE_NAME_FIREBASE, null);

            if (firebaseToken != null)
            {
                return firebaseToken;
            }
            else
            {
                FirebaseService firebaseService = new FirebaseService();
                firebaseService.RefreshToken();
            }

            return Preferences.GetString(PREFERENCE_NAME_FIREBASE, null);
        }

        public void SetDeviceId(string deviceId)
        {
            Preferences.Edit().PutString(PREFERENCE_NAME_FIREBASE, deviceId).Apply();
        }

        public void RefreshDeviceId()
        {
            FirebaseService firebaseService = new FirebaseService(Context);
            firebaseService.RefreshToken();
        }

        public string GetAmazonUserId()
        {
            return Preferences.GetString(PREFERENCE_NAME_AMAZON_USER_ID, null);
        }

        public void SetAmazonUserId(string userId)
        {
            Preferences.Edit().PutString(PREFERENCE_NAME_AMAZON_USER_ID, userId).Apply();
        }

        public string GetDeviceName()
        {
            return Preferences.GetString(PREFERENCE_NAME_DEVICE_NAME, null);
        }

        public void SetDeviceName(string deviceName)
        {
            Preferences.Edit().PutString(PREFERENCE_NAME_DEVICE_NAME, deviceName).Apply();
        }

        public string GetUserId()
        {
            return Preferences.GetString(PREFERENCE_NAME_ALEXA_USER_ID, null);
        }

        public void SetUserId(string userId)
        {
            Preferences.Edit().PutString(PREFERENCE_NAME_ALEXA_USER_ID, userId).Apply();
        }
    }
}

File: AndroidApp/app/src/main/java/com/jamespfluger/alexadevicefinder/utilities/PreferencesManager.java
package com.jamespfluger.alexadevicefinder.utilities;

import android.content.Context;
import android.content.SharedPreferences;

import com.jamespfluger.alexadevicefinder.notifications.FirebaseService;

public class PreferencesManager {
    private final String PREFERENCES_NAME = "com.jamespfluger.alexadevicefinder.SHARED_PREFERENCES";
    private final String PREFERENCE_NAME_FIREBASE = "firebasetoken";
    private final String PREFERENCE_NAME_AMAZON_USER_ID = "amazonuserid";
    private final String PREFERENCE_NAME_ALEXA_USER_ID = "alexauserid";
    private final String PREFERENCE_NAME_DEVICE_NAME = "devicename";
    private final SharedPreferences preferences;
    private final Context context;

    public PreferencesManager(Context context) {
        this.preferences = context.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE);
        this.context = context;
    }

    public String getDeviceId() {
        String firebaseToken = preferences.getString(PREFERENCE_NAME_FIREBASE, null);

        if (firebaseToken != null) {
            return firebaseToken;
        } else {
            FirebaseService firebaseService = new FirebaseService();
            firebaseService.refreshToken();
        }

        return preferences.getString(PREFERENCE_NAME_FIREBASE, null);
    }

    public void setDeviceId(String deviceId) {
        preferences.edit().putString(PREFERENCE_NAME_FIREBASE, deviceId).apply();
    }

    public void refreshDeviceId() {
        FirebaseService firebaseService = new FirebaseService(context);
        firebaseService.refreshToken();
    }

    public String getAmazonUserId() {
        return preferences.getString(PREFERENCE_NAME_AMAZON_USER_ID, null);
    }

    public void setAmazonUserId(String userId) {
        preferences.edit().putString(PREFERENCE_NAME_AMAZON_USER_ID, userId).apply();
    }

    public String getDeviceName() {
        return preferences.getString(PREFERENCE_NAME_DEVICE_NAME, null);
    }

    public void setDeviceName(String deviceName) {
        preferences.edit().putString(PREFERENCE_NAME_DEVICE_NAME, deviceName).apply();
    }

    public String getUserId() {
        return preferences.getString(PREFERENCE_NAME_ALEXA_USER_ID, null);
    }

    public void setUserId(String userId) {
        preferences.edit().putString(PREFERENCE_NAME_ALEXA_USER_ID, userId).apply();
    }
}
